tests/test_non_uniform_lattice_geometry/test_non_uniform_lattice_geometry.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+import os
+import sys
+sys.path.insert(0, os.pardir)
+sys.path.insert(0, os.path.join(os.pardir, 'openmoc'))
+import openmoc
+from testing_harness import TestHarness
+from input_set import NonUniformLatticeInput
+
+
+class NonUniformLatticeTestHarness(TestHarness):
+    """An eigenvalue calculation in a 3D Non-uniform Lattice with 7-group C5G7 data."""
+
+    def __init__(self):
+        super(NonUniformLatticeTestHarness, self).__init__()
+        self.input_set = NonUniformLatticeInput()
+        self.num_polar = 4
+        self.azim_spacing = self.spacing
+        self.z_spacing = 0.5
+        self.tolerance = 1E-4
+
+
+    def _create_geometry(self):
+        """Initialize CMFD and add it to the Geometry."""
+
+        super(NonUniformLatticeTestHarness, self)._create_geometry()
+
+        # Initialize CMFD
+        cmfd = openmoc.Cmfd()
+        cmfd.setCMFDRelaxationFactor(0.7)
+        #cmfd.setLatticeStructure(17,17)
+        cmfd.setWidths([[0.05,1.26,1.26,0.05], [0.05,1.26,1.26,0.05], [1.,1.5]])
+        cmfd.setGroupStructure([[1,2,3],[4,5,6,7]])
+        cmfd.setCentroidUpdateOn(False)
+
+        # Add CMFD to the Geometry
+        self.input_set.geometry.setCmfd(cmfd)
+        self.input_set.geometry.dumpToFile('non-uniform-lattice3.geo', non_uniform_lattice=True)
```

Comment:
is this dumped here to re-use in another test ? If so, we might want a comment in that test to remember

Reply:
It is. But no need to keep here. deleted